### ISSUE TITLE
Flow do not report error when use undefined property in condition

### ISSUE THREAD

**Issue by rlz** [facebook/flow#5655]
I expect an error in this code:

```js
// @flow

type Foo = {|
    foo: string
|}

function test(foo: Foo) {
    if (foo.bar) {
        console.log("WTF?")
    }
}
```

But flow thinks that the code is perfect:

https://flow.org/try/#0PTAEAEDMBsHsHcBQiAuBPADgU1AMVrKALygDeAPoqNaJAQFygDOKATgJYB2A5ouQL7JIAV04BjFO1idQKLCwAUdWI3ywAlGSo12kUEoIA6AEYBDVptLaaNMdKaxoWQ3G4KARAHUAKrgD87urWoIKCQA

**Comment by rlz** [comment]
After some digging I found this:

https://github.com/facebook/flow/issues/5429

The comment https://github.com/facebook/flow/issues/5429#issuecomment-352279788:

> Flow allows you to test for properties like this in conditionals. It's a common pattern for property testing and also refinements. Consider a union of exact objects; the property test can refine the union to one branch or the other.

doesn't seem reasonable to me. When I test union I know that variable I test has one common property and I'm testing that property, while in my example I already know that this property doesn't exist. If I want to check and use completely unknown property I should cast to `any` first.

Current behavior allows many misprint errors, which Flow suppose to catch in my opinion.

**Comment by CapacitorSet** [comment]
+1, I also ran into this error and would prefer it to be fixed.

I think a reasonable solution would be erroring on unknown properties in [exact object types](https://flow.org/en/docs/types/objects/#toc-exact-object-types), where Flow can prove that there are no additional properties.

**Comment by apsavin** [comment]
@rlz @CapacitorSet fixed in 0.68

**Comment by TrySound** [comment]
This probably can be closed.
/cc @vkurchatkin
